src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2426,45 +2528,187 @@ ostream& PgScrubber::show(ostream& out) const
+      break;
+    case cx_hash("scrubdebug_clr"):
+      op_cmd = asok_op_cmd_t::clear;
+      break;
+    default:
+      break;
+  }
+
+  for (auto& e : PgScrubber::asok_dispatch_tbl) {
+    if (e.op_cmd != op_cmd) {
+      continue;
+    }
+    if (e.cmd_hash != cmd_hash) {
+      continue;
+    }
+    if (e.param_hash && e.param_hash != param_hash) {
+      continue;
+    }
+    return (this->*(e.fn))(f, param, arg1, e.generated_param);
+  }
+  return -EINVAL;
+}
 
-  } else if (cmd == "unblock") {
-    // send an 'unblock' event, as if a blocked range was freed
+int PgScrubber::dbg_block_range(
+    [[maybe_unused]] Formatter* f,
+    [[maybe_unused]] std::string_view param,
+    [[maybe_unused]] std::string_view value,
+    [[maybe_unused]] int gen_param)
+{
+  // 'm_debug_blockrange' causes the next 'select_range' to report a blocked
+  // object
+  if (gen_param) {
+    m_debug_blockrange = 10;  // >1, so that will trigger fast state reports
+    f->dump_string("op", "blocking scrub range");
+  } else {
     m_debug_blockrange = 0;
     m_fsm->process_event(Unblocked{});
+    f->dump_string("op", "scrub range unblocked");
+  }
+  return 0;
+}
 
-  } else if ((cmd == "set") || (cmd == "unset")) {
-
-    if (param == "sessions") {
-      // set/reset the inclusion of the scrub sessions counter in 'query' output
-      m_publish_sessions = (cmd == "set");
-
-    } else if (param == "block") {
-      if (cmd == "set") {
-	// set a flag that will cause the next 'select_range' to report a
-	// blocked object
-	m_debug_blockrange = 10;  // >1, so that will trigger fast state reports
-      } else {
-	// send an 'unblock' event, as if a blocked range was freed
-	m_debug_blockrange = 0;
-	m_fsm->process_event(Unblocked{});
-      }
+int PgScrubber::dbg_unblock_range(
+    [[maybe_unused]] Formatter* f,
+    [[maybe_unused]] std::string_view param,
+    [[maybe_unused]] std::string_view value,
+    [[maybe_unused]] int gen_param)
+{
+  m_debug_blockrange = 0;
+  m_fsm->process_event(Unblocked{});
+  f->dump_string("op", "scrub range unblocked");
+  return 0;
+}
+
+int PgScrubber::dbg_set_sessions(
+    [[maybe_unused]] Formatter* f,
+    [[maybe_unused]] std::string_view param,
+    [[maybe_unused]] std::string_view value,
+    [[maybe_unused]] int gen_param)
+{
+  // set/reset the inclusion of the scrub sessions counter in 'query' output
+  m_publish_sessions = gen_param ? true : false;
+  f->dump_string("op", "ok");
+  return 0;
+}
+
+int PgScrubber::dbg_clear_sessions(
+    [[maybe_unused]] Formatter* f,
+    [[maybe_unused]] std::string_view param,
+    [[maybe_unused]] std::string_view value,
+    [[maybe_unused]] int gen_param)
+{
+  m_publish_sessions = false;
+  f->dump_string("op", "ok");
+  return 0;
+}
+
+int PgScrubber::dbg_reserv_delay(
+    [[maybe_unused]] Formatter* f,
+    [[maybe_unused]] std::string_view param,
+    [[maybe_unused]] std::string_view value,
+    [[maybe_unused]] int gen_param)
+{
+  milliseconds delay{4s};
```

Comment:
Might be worth adding this as a configurable. (default_reserv_delay)

Reply:
Don't think so, in this case. It's a default for a debug value. There should be some limit to configurability.

Reply:
Agreed. If there is no real need, let's hard-code.